bee-network/bee-gossip/src/peer/list.rs

```diff
@@ -77,29 +78,21 @@ impl PeerList {
         }
     }
 
-    pub fn insert_peer(&mut self, peer_id: PeerId, peer_info: PeerInfo) -> Result<(), (PeerId, PeerInfo, Error)> {
+    pub fn add(&mut self, peer_id: PeerId, peer_info: PeerInfo) -> Result<(), (PeerId, PeerInfo, Error)> {
         if self.contains(&peer_id) {
             return Err((peer_id, peer_info, Error::PeerIsDuplicate(peer_id)));
         }
 
         // Since we already checked that such a `peer_id` is not yet present, the returned value is always `None`.
-        let _ = self.peers.insert(peer_id, (peer_info, PeerState::Disconnected));
-
-        Ok(())
-    }
-
-    pub fn insert_local_addr(&mut self, addr: Multiaddr) -> Result<(), (Multiaddr, Error)> {
-        if self.local_addrs.contains(&addr) {
-            return Err((addr.clone(), Error::AddressIsDuplicate(addr)));
-        }
-
-        let _ = self.local_addrs.insert(addr);
+        let _ = self
+            .peers
+            .insert(peer_id, (peer_info, PeerState::default(), PeerMetrics::default()));
 
         Ok(())
     }
 
     pub fn remove(&mut self, peer_id: &PeerId) -> Result<PeerInfo, Error> {
```

Comment:
API is a tiny bit uncommon, you'd usually return an Option<PeerInfo>

Reply:
Yes, you're right, and I kept this API only to be consistent with the old code. I will be much more strict with a common API in #1174, so hopefully no surprises there :smile: